cmake/thirdParties.cmake

```diff
@@ -0,0 +1,21 @@
+if(MATIO_WITH_HDF5)
+    find_package(HDF5 REQUIRED)
```

Comment:
Why are hdf5 and zlib required dependencies? They are optional actually. If not found, the build should still succeed.

Reply:
I added it so you have to explicitly disable it with `MATIO_WITH_HDF5=0` and/or `MATIO_WITH_ZLIB=0` if you don't want them.
I will remove `REQUIRED` so it will continue to build.